WordPress/Classes/ViewRelated/Aztec/Processors/DocumentUploadProcessor.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+import Aztec
+
+/**
+ Manages the completion of a document upload, by replacing a placeholder attachment image with a
+ link to the uploaded document.
+ */
+class DocumentUploadProcessor: Processor {
+
+    private let mediaUploadID: String
+    private let remoteURLString: String
+    private let title: String
+
+    private lazy var processor = HTMLProcessor(tag: "img", replacer: { [remoteURLString, title] img in
+        var html = "<a href=\"\(remoteURLString)\">\(title)</a>"
```

Comment:
You are missing a check here. If you see the code on `ImgUploadProcessor` you will see that we do this:

```
guard let imgUploadID = img.attributes.named[MediaAttachment.uploadKey], self.mediaUploadID == imgUploadID else {
            return nil
        }
```

This make sure we changing the correct img link, this way you are going to replace any img in the post with the link info.